frontend/src/app/components/youtube/video-upload/video-upload.component.ts

```diff
@@ -0,0 +1,145 @@
+  participantId: number;
+  uploadButtonDisabled: boolean;
+
+  constructor(
+    private videoService: VideoService,
+    private scriptService: ScriptService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private ngxNotificationService: NgxNotificationService,
+    private crud: CrudService,
+    private errorHandler: ErrorHandlerService,
+    @Inject(DOCUMENT) document) { }
+
+  ngOnInit() {
+
+    this.uploadButtonDisabled = false;
+
+    this.videoService.currentVideosArray.subscribe(videosArr => this.videosArr = videosArr);
+    this.videoService.currentQuestionResponsesArray
+      .subscribe(questionResponsesArr => this.questionResponsesArr = questionResponsesArr);
+
+    this.scriptService.load('googleAjax', 'jsClient', 'uploadVideo', 'corsUpload').then(data => {
+      console.log('script loaded ', data);
+    }).catch(error => console.log(error));
+
+    this.experimentId = parseInt(this.route.snapshot.paramMap.get('experiment_id'), 10);
+    this.participantId = parseInt(this.route.snapshot.paramMap.get('participant_id'), 10);
+  }
+
+  /**
+   * Method that calls the YouTube upload for all videos from the current participant's
+   * responses to the experiment's questions.
+   */
+  callYouTubeUpload() {
+
+    if (this.videosArr != null) {
+
+      for (let i = 0; i < this.videosArr.length; i++) {
+
+        const questionId = this.questionResponsesArr[i].question_id;
+
+        // Assign the name of the video according to the experiment id, the question
+        // id and the participant id.
+        const videoName = 'experiment' + this.experimentId
+          + '_question' + questionId + '_participant' + this.participantId;
+
+        const videoDescription = 'Video corresponding to the response of participant ' + this.participantId
+          + ' of the question ' + questionId + ' of the experiment ' + this.experimentId + '.\n\nVideo uploaded by GestureWeb.';
+
+        // Call the upload. This executes a function in 'assets/js/upload_video.js'
+        const upload = new uploadToYouTube(this.videosArr[i], questionId, videoName, videoDescription);
+      }
+
+      this.uploadButtonDisabled = true;
+      this.sendNotification('The videos are being uploaded to YouTube', 'success', 'bottom-right');
+    }
+  }
+
+  /**
+   * Method that inserts to the database the participant's question responses
+   */
+  createQuestionResponses() {
+
+    // First, assign the video IDs to the QuestionResponse instances
+    // in the questionResponsesArr. As JQuery updated those strings,
+    // after the YouTube uploads finished, before this call to
+    // 'assignQuestionResponses()', the attribute 'videoId' of each
+    // QuestionResponse instance in that array is empty.
+    this.assignQuestionResponses();
+
+    // Once the video IDs have been set, everything's ready for
+    // inserting the question responses to the database.
+
+    for (let i = 0; i < this.questionResponsesArr.length; i++) {
```

Comment:
This seems like good enough for now. Maybe in the future we can try having a bulkCreate endpoint for the QuestionResponses model.

Reply:
I agree. A bulkCreate would be better. Thanks.